pkg/controller/zero-trust-workload-identity-manager/controller.go

```diff
@@ -169,19 +173,26 @@ type ZeroTrustWorkloadIdentityManagerReconciler struct {
 // +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=route.openshift.io,resources=routes/custom-host,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=events;secrets,verbs=get;list;watch
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operatorconditions,verbs=get;list;watch
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=operatorconditions/status,verbs=get;update;patch
 
 // New returns a new Reconciler instance.
 func New(mgr ctrl.Manager) (*ZeroTrustWorkloadIdentityManagerReconciler, error) {
 	c, err := customClient.NewCustomClient(mgr)
 	if err != nil {
 		return nil, err
 	}
+	operatorConditionName := os.Getenv("OPERATOR_CONDITION_NAME")
+	if operatorConditionName == "" {
+		return nil, errors.New("OPERATOR_CONDITION_NAME must be set")
+	}
 	return &ZeroTrustWorkloadIdentityManagerReconciler{
-		ctrlClient:    c,
-		ctx:           context.Background(),
-		eventRecorder: mgr.GetEventRecorderFor(utils.ZeroTrustWorkloadIdentityManagerControllerName),
-		log:           ctrl.Log.WithName(utils.ZeroTrustWorkloadIdentityManagerControllerName),
-		scheme:        mgr.GetScheme(),
+		ctrlClient:            c,
+		ctx:                   context.Background(),
+		eventRecorder:         mgr.GetEventRecorderFor(utils.ZeroTrustWorkloadIdentityManagerControllerName),
+		log:                   ctrl.Log.WithName(utils.ZeroTrustWorkloadIdentityManagerControllerName),
+		scheme:                mgr.GetScheme(),
+		operatorConditionName: operatorConditionName,
```

Comment:
Can you add a comment on why operatorConditionName is not validated for existence before setting it to ZeroTrustWorkloadIdentityManagerReconciler? Also, validation for the name string should be applied for valid object name strings.